http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -185,13 +194,50 @@ public JettyHttpClient(
             transport = new HttpClientTransportOverHTTP(CLIENT_TRANSPORT_SELECTORS);
         }
 
-        if (authenticationEnabled) {
+        if (authenticationEnabled && config.getBasicAuthUser() == null && config.getBasicAuthPass() == null) {
```

Comment:
should the second `&&` be a `||`, since we don't want the username or the password to be null?